### PR TITLE
Catch the exception that we get when there are no updates to apply

### PR DESCRIPTION
**Pull request by sihil**
This changes the error you get when there are no changes:
screen shot 2014-12-24 at 21 01 06

Into a successful deploy:
screen shot 2014-12-24 at 21 01 25

There is currently a bug in which the `CheckUpdateEventsTask` task doesn't really work and if fixed might break this again - but this will work for now.

### CODE REVIEW

File: magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala [hunk 1]
```diff
@@ -27,7 +28,12 @@ case class UpdateCloudFormationTask(cloudFormationStackName: String, template: P
     MessageBroker.info(s"Parameters: $actualParameters")
 
     if (CloudFormation.describeStack(cloudFormationStackName).isDefined)
-      CloudFormation.updateStack(cloudFormationStackName, template.string, actualParameters)
+      try {
+        CloudFormation.updateStack(cloudFormationStackName, template.string, actualParameters)
+      } catch {
+        case ase:AmazonServiceException if ase.getMessage contains "No updates are to be performed." =>
```
Comment (philwills) [inline magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala]: I'm guessing you looked at the errorCode on this and it wasn't helpful.

Comment (sihil) [inline magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala]: The HTTP status code or an amazon error code? If the latter, I didn't notice it existed.

Comment (philwills) [inline magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala]: I was thinking Amazon error code, which there seems to be a method for on `AmazonServiceException`

Comment (sihil) [inline magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala]: Ah - good spot. I'll try and revisit that - that'd certainly be a more robust approach.

Comment (sihil) [inline magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala]: I think that the error code returns a pretty generic value from the failure examples in Riff-Raff in CODE. e.g. http://riffraff.code.dev-gutools.co.uk/deployment/view/edf42137-200b-4021-8925-650881a79255
Doesn't seem to be a specific identifiable code that is likely to be unique.

Comment (philwills) [inline magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala]: :disappointed:

**Pull request merged**